### PR TITLE
Add multiModel field to ServingRuntime spec

### PR DESCRIPTION
**Pull request by pvaneck**
This `multiModel` field will be used to define if a model is meant for SMS vs MMS.

Related: https://github.com/kserve/modelmesh-serving/pull/89

### CODE REVIEW

File: config/crd/serving.kserve.io_clusterservingruntimes.yaml [hunk 1]
```diff
@@ -653,6 +653,8 @@ spec:
                 type: string
               httpDataEndpoint:
                 type: string
+              multiModel:
+                type: boolean
```
Comment (animeshsingh) [inline config/crd/serving.kserve.io_clusterservingruntimes.yaml:657]: Can it be "colocate"? Not sure -we understand single/multi-model literature well, but at the basic level it's a decision to colocate or not?

File: pkg/apis/serving/v1beta1/predictor_model.go [hunk 2]
```diff
@@ -115,14 +115,14 @@ func (m *ModelSpec) GetSupportingRuntimes(cl client.Client, namespace string, is
...
-		if !rt.Spec.IsDisabled() && isRuntimeModelMeshCompatible(&rt.Spec) == isMMS && m.RuntimeSupportsModel(&rt.Spec) {
+		if !rt.Spec.IsDisabled() && rt.Spec.IsMultiModelRuntime() == isMMS && m.RuntimeSupportsModel(&rt.Spec) {
 			srSpecs = append(srSpecs, rt.Spec)
 		}
 	}
 
 	for i := range clusterRuntimes.Items {
 		crt := &clusterRuntimes.Items[i]
-		if !crt.Spec.IsDisabled() && isRuntimeModelMeshCompatible(&crt.Spec) == isMMS && m.RuntimeSupportsModel(&crt.Spec) {
+		if !crt.Spec.IsDisabled() && crt.Spec.IsMultiModelRuntime() == isMMS && m.RuntimeSupportsModel(&crt.Spec) {
```
Comment (animeshsingh) [inline pkg/apis/serving/v1beta1/predictor_model.go:125]: Looking at this, maybe `multiModel` makes sense

Comment (pvaneck) [reply pkg/apis/serving/v1beta1/predictor_model.go:125]: Yea, pretty much just need this so ModelMesh knows whether to create deployments for a ServingRuntime or not.

**Pull request merged**